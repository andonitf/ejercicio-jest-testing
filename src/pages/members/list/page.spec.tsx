import * as React from 'react';
import { mount } from 'enzyme';
import { MemberListPage } from './page';
import { Member } from './viewModel';

describe('pages/members/list/page specs', () => {
  it('should render the MemberListPage component as expected when passing required properties', () => {

    // Arrange
    const members: Member[] = [
      {
        id: 1,
        name: 'Login 1',
        avatarUrl: 'http://fake-test-url/member/1',
      },
      {
        id: 2,
        name: 'Login 2',
        avatarUrl: 'http://fake-test-url/member/2',
      },
    ];
    const props = { members };

    // Act
    const component = mount(<MemberListPage members={props.members} />);

    // Assert
    expect(component).toMatchSnapshot();
  });
});
