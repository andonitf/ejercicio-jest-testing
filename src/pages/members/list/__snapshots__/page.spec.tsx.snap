// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/members/list/page specs should render the MemberListPage component as expected when passing required properties 1`] = `
<MemberListPage
  members={
    Array [
      Object {
        "avatarUrl": "http://fake-test-url/member/1",
        "id": 1,
        "name": "Login 1",
      },
      Object {
        "avatarUrl": "http://fake-test-url/member/2",
        "id": 2,
        "name": "Login 2",
      },
    ]
  }
>
  <h2>
    Members
  </h2>
  <Table
    members={
      Array [
        Object {
          "avatarUrl": "http://fake-test-url/member/1",
          "id": 1,
          "name": "Login 1",
        },
        Object {
          "avatarUrl": "http://fake-test-url/member/2",
          "id": 2,
          "name": "Login 2",
        },
      ]
    }
  >
    <table
      className="table table-striped"
    >
      <Component>
        <thead>
          <tr>
            <th>
              Avatar
            </th>
            <th>
              Id
            </th>
            <th>
              Name
            </th>
          </tr>
        </thead>
      </Component>
      <Body
        members={
          Array [
            Object {
              "avatarUrl": "http://fake-test-url/member/1",
              "id": 1,
              "name": "Login 1",
            },
            Object {
              "avatarUrl": "http://fake-test-url/member/2",
              "id": 2,
              "name": "Login 2",
            },
          ]
        }
      >
        <tbody>
          <Row
            key="1"
            member={
              Object {
                "avatarUrl": "http://fake-test-url/member/1",
                "id": 1,
                "name": "Login 1",
              }
            }
          >
            <tr>
              <td>
                <img
                  src="http://fake-test-url/member/1"
                  style={
                    Object {
                      "maxWidth": "10rem",
                    }
                  }
                />
              </td>
              <td>
                <span>
                  1
                </span>
              </td>
              <td>
                <span>
                  Login 1
                </span>
              </td>
            </tr>
          </Row>
          <Row
            key="2"
            member={
              Object {
                "avatarUrl": "http://fake-test-url/member/2",
                "id": 2,
                "name": "Login 2",
              }
            }
          >
            <tr>
              <td>
                <img
                  src="http://fake-test-url/member/2"
                  style={
                    Object {
                      "maxWidth": "10rem",
                    }
                  }
                />
              </td>
              <td>
                <span>
                  2
                </span>
              </td>
              <td>
                <span>
                  Login 2
                </span>
              </td>
            </tr>
          </Row>
        </tbody>
      </Body>
    </table>
  </Table>
</MemberListPage>
`;
